libbeat/esleg/eslegclient/connection.go

```diff
@@ -435,8 +443,8 @@ func (conn *Connection) execHTTPRequest(req *http.Request) (int, []byte, error)
 		req.SetBasicAuth(conn.Username, conn.Password)
 	}
 
-	if conn.APIKey != "" {
-		req.Header.Add("Authorization", "ApiKey "+conn.APIKey)
+	if conn.encodedAPIKey != "" {
+		req.Header.Add("Authorization", "ApiKey "+conn.encodedAPIKey)
```

Comment:
nit: We still have an unnecessary alloc + copy here. If we initialize `encodedAPIKey = "ApiKey " + ...`, then we can just pass it to `Header` as is.

Reply:
Changed in 47c9aacaaa04e9cc1e304d517144386a77483e02.